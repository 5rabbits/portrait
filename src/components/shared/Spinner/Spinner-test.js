import { React, toJSON, mount } from 'test/helper'
import Spinner from 'shared/Spinner'

describe('Spinner', () => {
  it('should render with minimum props', () => {
    const component = mount(
      <Spinner />,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  describe('if props.backgroundColor is defined but props.color is not', () => {
    it('should add the class "Spinner--force-color"', () => {
      const component = mount(
        <Spinner
          backgroundColor="#f00"
        />,
      )

      expect(toJSON(component)).toMatchSnapshot()
    })
  })
})
