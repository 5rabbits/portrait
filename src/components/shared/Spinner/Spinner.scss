@import '~styles/core';

$background-color: rgba(#000, 0.05);
$foreground-color: $primary-color;

.Spinner {
  display: inline-block;
  height: 1em;
  width: 1em;
  vertical-align: baseline;
  border-radius: 50%;
  font-size: 1em;
  position: relative;
  border-style: solid;
  border-width: 0.1em;
  border-color: $background-color;
  border-left-color: $foreground-color;
  transform: translateZ(0);
  animation: spin 1s infinite linear;
}

.Spinner--force-color {
  border-left-color: $foreground-color !important;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
