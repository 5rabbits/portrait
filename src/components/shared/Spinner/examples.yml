-
  description: A simple spinner.
  code: >
    <Spinner />

-
  description: Spinner with text.
  code: >
    <span>
      <Spinner /> Some text
    </span>

-
  description: The spinner scales with the font size by default.
  code: >
    <span style={{ fontSize: 100 }}>
      <Spinner /> Some text
    </span>

-
  description: Custom colors and sizes.
  code: >
    <Spinner
      backgroundColor="#000"
      color="#FF5370"
      ringSize={2}
      size={100}
    />
