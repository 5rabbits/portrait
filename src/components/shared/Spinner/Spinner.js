import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './Spinner.scss'

export default class Spinner extends PureComponent {
  static propTypes = {
    backgroundColor: PropTypes.string,
    color: PropTypes.string,
    ringSize: PropTypes.oneOf([
      PropTypes.number, PropTypes.string,
    ]),
    size: PropTypes.oneOf([
      PropTypes.number, PropTypes.string,
    ]),
  }

  static defaultProps = {
    backgroundColor: null,
    color: null,
    size: null,
    ringSize: null,
  }

  render() {
    const { backgroundColor, color, ringSize, size } = this.props

    return (
      <div
        className={cx('Spinner', {
          'Spinner--force-color': backgroundColor && !color,
        })}
        style={{
          height: size,
          width: size,
          borderColor: backgroundColor,
          borderLeftColor: color,
          borderWidth: ringSize,
        }}
      />
    )
  }
}
