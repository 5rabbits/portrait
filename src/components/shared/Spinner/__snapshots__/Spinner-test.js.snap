// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner if props.backgroundColor is defined but props.color is not should add the class "Spinner--force-color" 1`] = `
<Spinner
  backgroundColor="#f00"
  color={null}
  ringSize={null}
  size={null}
>
  <div
    className="Spinner Spinner--force-color"
    style={
      Object {
        "borderColor": "#f00",
        "borderLeftColor": null,
        "borderWidth": null,
        "height": null,
        "width": null,
      }
    }
  />
</Spinner>
`;

exports[`Spinner should render with minimum props 1`] = `
<Spinner
  backgroundColor={null}
  color={null}
  ringSize={null}
  size={null}
>
  <div
    className="Spinner"
    style={
      Object {
        "borderColor": null,
        "borderLeftColor": null,
        "borderWidth": null,
        "height": null,
        "width": null,
      }
    }
  />
</Spinner>
`;
